specs/experimental/op-stack-manager.md

```diff
@@ -0,0 +1,261 @@
+This corresponds to the L1 ProxyAdmin owner and L2 ProxyAdmin owner [admin roles].
+- `systemConfigInputs` is a struct defining configuration of the `SystemConfig` inputs.
+- `l2OutputOracleInputs` is a struct defining configuration of the `L2OutputOracle` inputs.
+
+The `l2ChainId` has the following restrictions:
+
+- It is limited to a `uint64`. This contradicts the [interop specification],
+which allows chain ID to be a `uint256`. This contradiction exists because it is
+currently unspecified how to unify a 256-bit chain ID with the existing batch inbox
+address convention, as tracked in [ethereum-optimism/specs#94]. This limitation
+may be removed in a future version of the OP Stack Manager once a solution to
+that issue is found.
+- It must not be equal to 0.
+- It must not be equal to the chain ID of the chain the OP Stack Manager is
+deployed on.
+- It must not be equal to a chain ID that is already present in the
+[ethereum-lists/chains] repository. This is not enforced onchain, but may matter
+for future versions of OP Stack Manager that handle upgrades via a privileged role.
+
+The `SystemConfigInputs` struct is defined as follows:
+
+```solidity
+struct SystemConfigInputs {
+  address systemConfigOwner;
+  uint256 overhead;
+  uint256 scalar;
+  bytes32 batcherHash;
+  address unsafeBlockSigner;
+}
+```
+
+Details on each parameter can be found in the code comments of the [`SystemConfig` contract].
+Note that not all `SystemConfig` parameters are able to be configured.
+This is because the OP Stack Manager enforces compliance with the [standard configuration], and only allows
+configuration where allowed by the standard configuration.
+
+The `L2OutputOracleInputs` struct is defined as follows:
+
+```solidity
+struct L2OutputOracleInputs {
+  address proposer;
+  address challenger;
+}
+```
+
+Details on each parameter can be found in the code comments of the [`L2OutputOracle` contract].
+Note that not all `L2OutputOracle` parameters are able to be configured, for the reason as the `SystemConfig`.
+
+On success, the following event is emitted:
+
+```solidity
+event Deployed(uint256 indexed l2ChainId, SystemConfig indexed systemConfig);
+```
+
+This method reverts on failure. This occurs when:
+
+- The input `l2ChainId` does not comply with the restrictions above.
+- The resulting configuration is not compliant with the [standard configuration].
+
+### Getter Methods
+
+The following interface defines the available getter methods:
+
+```solidity
+/// @notice The logic address and initializer selector for an implementation contract.
+struct Implementation {
+  address logic; // Address containing the deployed logic contract.
+  bytes4 initializer; // Function selector for the initializer.
+}
+
+/// @notice Returns the latest approved release of the OP Stack contracts.
+/// @notice Release strings follow semver and are named with the
+/// format `op-contracts/vX.Y.Z`.
+function latestRelease() external view returns (bytes32);
```

Comment:
`bytes32` is a weird return type for a semver string.

Reply:
I often use `bytes32` for short strings in solidity because it's a fixed size type so it can live on the stack and is easier to work with, whereas `string` is not fixed size so it must either be calldata/memory/storage

I am open to changing this though, curious if @tynes has any preferences?

Reply:
Slight preference for `string` as its going to be human readable, ie etherscan. If it was machine used then I think `bytes32` is better (like custom gas token)

Reply:
Updated to `string` in 8a35fee